<launch>
  <!-- Arguments -->
  <arg name="robot_name" value="cube_stack_arm"/>
  <arg name="model" default="$(find cube_stack_env)/urdf/cube_stack_arm.xacro"/>

  <!-- Set essential parameters -->
  <param name="/cube_stack_arm/dynamixel_ids" value="1,2,3,4,5,"/>
  <param name="/cube_stack_arm/robot_description"
        command="xacro $(arg model)"/>

  <!--Setup TF -->
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" ns="/cube_stack_arm">
    <param name="publish_frequency" type="double" value="30.0" />
    <remap from="/joint_states" to="/cube_stack_arm/joint_states"/>
    <remap from="/robot_description" to="/cube_stack_arm/robot_description" />
  </node>

  <!-- Launch Dynamixel controller -->
  <node pkg="cube_stack_env" type="sync_read_write_node" name="sync_read_write_node"/>

</launch>
